be/src/exprs/table_function/unnest.h

```diff
@@ -50,17 +50,28 @@ class Unnest final : public TableFunction {
 
             for (int row_idx = 0; row_idx < nullable_array_column->size(); ++row_idx) {
                 if (nullable_array_column->is_null(row_idx)) {
+                    if (state->get_is_left_join()) {
+                        // to support unnest with null.
+                        compact_offset -= 1;
+                        compacted_array_elements->append_nulls(1);
+                    }
                     compact_offset +=
                             offset_column->get(row_idx + 1).get_int32() - offset_column->get(row_idx).get_int32();
                     int32 offset = offset_column->get(row_idx + 1).get_int32();
                     compacted_offset_column->append_datum(offset - compact_offset);
                 } else {
+                    if (offset_column->get(row_idx + 1).get_int32() == offset_column->get(row_idx).get_int32() &&
+                        state->get_is_left_join()) {
+                        // to support unnest with null.
+                        compact_offset -= 1;
+                        compacted_array_elements->append_nulls(1);
+                    } else {
+                        compacted_array_elements->append(
+                                *(col_array->elements_column()), offset_column->get(row_idx).get_int32(),
+                                offset_column->get(row_idx + 1).get_int32() - offset_column->get(row_idx).get_int32());
+                    }
                     int32 offset = offset_column->get(row_idx + 1).get_int32();
                     compacted_offset_column->append_datum(offset - compact_offset);
```

Comment:
to simpified the branch? the offset is hard to understant...

and add `what are you doing?` to describe this pr?

Reply:
Yes, to simplified the branch, compact_offset means the diff between result offset and original offset, compact_offset -= 1 means we added a null value, and the result offset should be one larger than original offset. 

and `what are you doing` is added.